scraper/src/libretexts2zim/client.py

```diff
@@ -30,6 +30,49 @@ class LibreTextsHome(BaseModel):
     welcome_image_url: str
 
 
+class DekiPage(BaseModel):
+    id: str
+    title: str
+    parent: "DekiPage | None" = None
+    children: list["DekiPage"] = []
+
+    def __repr__(self) -> str:
+        return (
+            f"DekiPage(id='{self.id}', title='{self.title}', "
+            f"parent='{'None' if not self.parent else self.parent.id}', "
+            f"children='{','.join([child.id for child in self.children])}')"
+        )
+
+    @property
+    def self_and_parents(self) -> list["DekiPage"]:
```

Comment:
I don't get your point, you want to replace `self_and_parents` with `Self`?

Reply:
No, the return type. I wonder why you're not using `Self` instead

Reply:
Simply because I didn't knew about it ^^

Now that I know about it, I can say that I don't want to use `Self` because I still don't fully understand all constraints regarding inheritance, pretty sure to get bitten by this at some point, and it is not like it is a big burden to use `"DekiPage"` (or `"LibraryPage"` now).

Reply:
I am talking about [`typing.Self`](https://docs.python.org/3/library/typing.html#typing.Self), not `self`. It's just a type hint that has the same meaning as what you did except it's the proper way to do it.